src/Shared/RoslynUtils/CodeWriter.cs

```diff
@@ -23,4 +24,30 @@ public void EndBlock()
         this.Indent--;
         this.WriteLine("}");
     }
+
+    public void EndBlockWithComma()
+    {
+        this.Indent--;
+        this.WriteLine("},");
+    }
+
+    // The IndentedTextWriter adds the indentation
+    // _after_ writing the first line of text. This
+    // method can be used ot initialize indentation
+    // when an emit method might only emit one line
+    // of code or when the code writer is emitting
+    // indented code as part of a larger string.
+    public void InitializeIndent()
+    {
+        if (_indentationInitialized)
```

Comment:
Why do we need this bool? Do we really call `InitializeIndent()` multiple times on a single CodeWriter?

Reply:
Ah, you're right! This is a vestige from when I was calling `InitializeIndent` within the loop inside `EmitQueryOrRouteResolver`.